### ISSUE TITLE
Feature Proposal: Linux: Set LD_LIBRARY_PATH too

### ISSUE THREAD

**Issue by jansenm** [jenkinsci/sqlplus-script-runner-plugin#40]
When installing oracle instant client on a machine with root rights its possible to add $ORACLE_HOME/lib to the linker configuration (eg. /etc/ld.so.conf.d/..) so sqlplus can find the shared libs.

Without root rights this happens

```session
$ $ORACLE_HOME/sqlplus
.../oracle-client/11.2.0.4.0/sqlplus: error while loading shared libraries: libsqlplus.so: cannot open shared object file: No such file or directory
```

Non root user have do do this:

```session
$ env LD_LIBRARY_PATH=$ORACLE_HOME $ORACLE_HOME/sqlplus
```

By adding ORACLE_HOME to the LD_LIBRARY_PATH variable you can run it without having to change your systems linker paths.

it also opens up the possibility to run more than one oracle instant client version on a host. Useful for example for a jenkins build machine.

So here is the proposal:

Add oracle instant client as a Global Tool (like mvn, java, etc). When calling sqlplus also add ORACLE_HOME to LD_LIBRARY_PATH. This would make it possible to install/define more than one oracle instant client version on the jenkins slave.

In the jobs you would just just the tool version you want.

This is just a proposal btw.

if the global tool configuration is to much work please consider adding ORACLE_HOME to LD_LIBRARY_PATH before calling sqlplus at least on linux.

**Comment by boaglio** [comment]
Hi, due to Oracles License I don't think I can do this installation process. 
And about LD_LIBRARY_PATH, the plugin already adds $ORACLE_HOME and $ORACLE_HOME/lib

**Comment by jansenm** [comment]
It does? Thats weird. Doesn't work for us. Then i guess i have to do some testing.

We extracted the instant client, set ORACLE_HOME and SQLPLUS_PATH (globally) and got the dreaded *shared library not found" problem.

**Comment by boaglio** [comment]
Turn _Debug Info_ on and you will be able to see LD_LIBRARY_PATH value on your next build.

**Comment by jansenm** [comment]
I did some testing. With debug output on i can see the plugin tries to set LD_LIBRARY_PATH and ORACLE_HOME

```log
Getting ORACLE_HOME...
Using custom SQL*Plus location
SQL*Plus >>> /usr/local/phx-unlimited/software/oracle-client/11.2.0.4.0/sqlplus
[DEBUG] Using custom ORACLE_HOME
--------------------------------------------------------------------------
Using custom ORACLE_HOME
 detected host = jenkins-dev
 slave machine ? true
--------------------------------------------------------------------------
Using ORACLE_HOME =/usr/local/phx-unlimited/software/oracle-client/11.2.0.4.0
--------------------------------------------------------------------------
Getting SQLPlus version
[DEBUG] ORACLE_HOME = /usr/local/phx-unlimited/software/oracle-client/11.2.0.4.0
[DEBUG] LD_LIBRARY_PATH = /usr/local/phx-unlimited/software/oracle-client/11.2.0.4.0/lib
SQL*Plus exec file = sqlplus
--------------------------------------------------------------------------
[DEBUG] Statement:
/usr/local/phx-unlimited/software/oracle-client/11.2.0.4.0/sqlplus -v  
--------------------------------------------------------------------------
[mjansen_test] $ /usr/local/phx-unlimited/software/oracle-client/11.2.0.4.0/sqlplus -v
/usr/local/phx-unlimited/software/oracle-client/11.2.0.4.0/sqlplus: error while loading shared libraries: libsqlplus.so: cannot open shared object file: No such file or directory
Process exited with status 127
--------------------------------------------------------------------------
 detected host = jenkins-dev
--------------------------------------------------------------------------
Using ORACLE_HOME =/usr/local/phx-unlimited/software/oracle-client/11.2.0.4.0
--------------------------------------------------------------------------
Running defined script on SYS_ADMIN/********@PHXULDB
Temp script: file:/usr/local/phx-unlimited/jenkins/putb06-heavy/home/workspace/mjansens_test/mjansen_test/temp-script-15603505931966313173655579584953.sql
--------------------------------------------------------------------------
[DEBUG] ORACLE_HOME = /usr/local/phx-unlimited/software/oracle-client/11.2.0.4.0
[DEBUG] LD_LIBRARY_PATH = /usr/local/phx-unlimited/software/oracle-client/11.2.0.4.0/lib:/usr/local/phx-unlimited/software/oracle-client/11.2.0.4.0
Master Work Directory = /usr/local/phx-unlimited/jenkins/putb06-heavy/home/workspace/mjansens_test/mjansen_test
[DEBUG]  Statement: 
--------------------------------------------------------------------------
/usr/local/phx-unlimited/software/oracle-client/11.2.0.4.0/sqlplus -L SYS_ADMIN/phxuadmin@PHXULDB @/usr/local/phx-unlimited/jenkins/putb06-heavy/home/workspace/mjansens_test/mjansen_test/temp-script-15603505931966313173655579584953.sql  
--------------------------------------------------------------------------
 
[mjansen_test] $ /usr/local/phx-unlimited/software/oracle-client/11.2.0.4.0/sqlplus -L SYS_ADMIN/phxuadmin@PHXULDB @/usr/local/phx-unlimited/jenkins/putb06-heavy/home/workspace/mjansens_test/mjansen_test/temp-script-15603505931966313173655579584953.sql
/usr/local/phx-unlimited/software/oracle-client/11.2.0.4.0/sqlplus: error while loading shared libraries: libsqlplus.so: cannot open shared object file: No such file or directory
Process exited with status 127
--------------------------------------------------------------------------
Exit code: 127
--------------------------------------------------------------------------
java.lang.RuntimeException: Process exited with error
	at org.jenkinsci.plugins.sqlplus.script.runner.SQLPlusRunner.run(SQLPlusRunner.java:444)
	at org.jenkinsci.plugins.sqlplus.script.runner.SQLPlusRunnerBuilder.perform(SQLPlusRunnerBuilder.java:174)
	at hudson.tasks.BuildStepCompatibilityLayer.perform(BuildStepCompatibilityLayer.java:79)
	at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:20)
	at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:741)
	at hudson.model.Build$BuildExecution.build(Build.java:206)
	at hudson.model.Build$BuildExecution.doRun(Build.java:163)
	at hudson.model.AbstractBuild$AbstractBuildExecution.run(AbstractBuild.java:504)
	at hudson.model.Run.execute(Run.java:1818)
	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
	at hudson.model.ResourceController.execute(ResourceController.java:97)
	at hudson.model.Executor.run(Executor.java:429)
ERROR: Process exited with error
```

But i noticed some things (and it doesn't work).

* At one point it sets LD_LIBRARY_PATH only to the $ORACLE_HOME/lib/ directory which wouldn't work for us.
* At another it sets both correctly but the call still fails.

I replaced sqlplus by a bash script that printed the content of both environment variables and got:

```
$ /usr/local/phx-unlimited/software/oracle-client/11.2.0.4.0/sqlplus -v
ORACLE_HOME=/usr/local/phx-unlimited/software/oracle-client/11.2.0.4.0
LD_LIBRARY_PATH=
```

It looks as if the variable is not exported/forwarded to sqlplus.

The configuration was completely done on job level btw.

**Comment by boaglio** [comment]
Thank you for your test! 
Fixed this issue on release **2.0.8** , please give it a try !
